# Generated by Django 2.2.12 on 2020-05-14 15:36

from django.db import migrations


def fill_validation_result(apps, schema_editor):
    """
    This data migration fills the result field on CAVLValidationTaskResult to make the result and error states more
    explicit. We need to migrate any existing CAVLValidationTaskResult which have failed on succeeed to have the correct
    result.
    """
    CAVLValidationTaskResult = apps.get_model("avl", "CAVLValidationTaskResult")

    # Assume any task that succeeded are valid AVLFeeds
    CAVLValidationTaskResult.objects.filter(status="SUCCESS").update(result="VALID")

    # Assume any task that failed failed due to a system error
    CAVLValidationTaskResult.objects.filter(status="FAILURE").update(
        result="SYSTEM_ERROR"
    )


class Migration(migrations.Migration):
    dependencies = [
        ("avl", "0003_cavlvalidationtaskresult_result"),
    ]

    operations = [
        migrations.RunPython(
            fill_validation_result, migrations.RunPython.noop, elidable=True
        )
    ]
